### PR TITLE
[image-picker] Fix cropping box stuck in center 

### LINKED ISSUE #1818: ImagePicker -- the image has some offsets on top on iOS
The image cannot be placed at the right place since there is offsets.
The issue happens on both launchCameraAsync and launchImagebriaryAsync.
Some area of launchCameraAsync cannot be dragged to.

### Environment

Environment:
  OS: macOS High Sierra 10.13.3
  Node: 8.11.2
  Yarn: Not Found
  npm: 6.1.0
  Watchman: Not Found
  Xcode: Xcode 9.3.1 Build version 9E501
  Android Studio: Not Found

Packages: (wanted => installed)
  expo: ^27.0.0 => 27.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-27.0.2.tar.gz => 0.55.2

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/kevinscience-e22fac89-c4ac-4213-9216-16132554e54b.tar.gz

iOS has the issue, Android is fine.

### Steps to Reproduce

```
let pickerResult = await Expo.ImagePicker.launchImageLibraryAsync({
allowsEditing: true,
aspect: [1, 1]});

 let pickerResult = await ImagePicker.launchCameraAsync({
            allowsEditing: true,
            aspect: [1, 1],
        });
```

### Expected Behavior

The image picker can pick anywhere of image with no offsets

### Actual Behavior

It has some offsets on both launchCameraAsync and launchImageibraryAsync.

launchCameraAsync:
giphy 1
  
launchImageLibraryAsync
giphy

### Reproducible Demo

https://snack.expo.io/rkpwDDCym

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolve #1818.

# How

Since iOS 6, exist bug, which causes that you can't move the cropping box.
I've finde a workaround here:
https://stackoverflow.com/questions/12630155/uiimagepicker-allowsediting-stuck-in-center
This method unlocks the editing box.

# Test Plan

I've tested https://snack.expo.io/@lukaszkosmaty/github---imagepicker-crop-stuck-in-center on different iOS versions. Everything seems to work fine.

### CODE REVIEW

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m [hunk 1]
```diff
@@ -481,4 +529,11 @@ - (UIImage *)fixOrientation:(UIImage *)srcImg {
   return mediaTypes;
 }
 
++ (void)foreachSubviewIn:(UIView *)view call:(void (^)(UIView *subview))function {
```
Comment (bbarthec) [inline packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m]:
```suggestion
+ (void)foreachSubviewIn:(UIView *)view call:(void (^)(UIView *subview))function
{
```

File: packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m [hunk 2]
```diff
@@ -125,7 +126,54 @@ - (void)launchImagePicker:(EXImagePickerTarget)target
...
+    [utils.currentViewController presentViewController:self.picker animated:YES completion:^{
+      [self freeCroppingBox:self.picker];
+    }];
+  });
+}
+
+// Since iOS 6, exist bug, which causes that you can't move the cropping box.
+// That's why we had to apply a workaround to unlock editing box.
+// You can read more about this here: https://stackoverflow.com/questions/12630155/uiimagepicker-allowsediting-stuck-in-center.
+- (void)freeCroppingBox:(UIImagePickerController *)imagePickerController {
```
Comment (bbarthec) [inline packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m]:
```suggestion
- (void)freeCroppingBox:(UIImagePickerController *)imagePickerController
{
```

**Review by bbarthec — APPROVED** [review]
LGTM 😉 Just two style suggestions 👅

File: packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m [hunk 3]
```diff
@@ -125,10 +126,49 @@ - (void)launchImagePicker:(EXImagePickerTarget)target
...
+        CGFloat inset = ABS(size.width - size.height) / 2;
+        self->_imageScrollView.contentInset = UIEdgeInsetsMake(inset, 0, inset, 0);
+        // 2. centering image by default
+        if (isNewImageScrollView) {
+            CGSize contentSize = self->_imageScrollView.contentSize;
+            if (contentSize.height > contentSize.width) {
+                CGFloat offset = round((contentSize.height - contentSize.width) / 2 - inset);
+                self->_imageScrollView.contentOffset = CGPointMake(self->_imageScrollView.contentOffset.x, offset);
+            }
+        }
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m]: Could you fix the indentation in this method?

**Review by tsapeta — APPROVED** [review]

**Pull request merged**